### ISSUE TITLE
Failed to conan install gRPC/1.8.3@inexorgame/stable 

### ISSUE THREAD

**Issue by jinq0123** [inexorgame/inexor-core#589]
`conan install gRPC/1.8.3@inexorgame/stable --build missing`

```
...
gRPC/1.8.3@inexorgame/stable: WARN: Trying to remove dirty source folder
gRPC/1.8.3@inexorgame/stable: WARN: This can take a while for big packages
ERROR: gRPC/1.8.3@inexorgame/stable: Error in source() method, line 40
        endif()''', '''  # CONAN: Use C-ARES from the submodule''')
        ConanException: replace_in_file didn't find pattern '  if(gRPC_INSTALL)
    message(WARNING "gRPC_INSTALL will be forced to FALSE because gRPC_CARES_PROVIDER is \"module\"")
    set(gRPC_INSTALL FALSE)
  endif()' in 'grpc-1.8.3/CMakeLists.txt' file.
```

**Comment by Croydon** [comment]
https://github.com/inexorgame/conan-grpc/commit/6901d2febb4a51e233961634ab9ab66a20dbde3f

@jinq0123 Please be aware that only `gRPC/1.1.0@inexorgame/stable` can be considered as actual stable right now.

I patched 1.8.3 so far that it builds successfully (tested on Windows), but consuming this version is untested for now. Could you maybe re-test and provide feedback?

**Comment by jinq0123** [comment]
Failed to install debug version with Windows VS2015:

```
λ conan install gRPC/1.8.3@inexorgame/stable -s build_type=Debug -s compiler.runtime=MDd --build missing
```
```
...
PROTOBUF_SRC_ROOT_FOLDER =
Protobuf_INCLUDE_DIR = C:/Users/jinqing/.conan/data/Protobuf/3.5.1/inexorgame/stable/package/b17b520b4b55729a7391c6b2d20631fec4cf1564/include
CMake Error at D:/Program Files/CMake/share/cmake-3.7/Modules/FindPackageHandleStandardArgs.cmake:138 (message):
  Could NOT find Protobuf (missing: Protobuf_LIBRARIES) (found version
  "3.5.1")
Call Stack (most recent call first):
  D:/Program Files/CMake/share/cmake-3.7/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  D:/Program Files/CMake/share/cmake-3.7/Modules/FindProtobuf.cmake:404 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  CMakeLists.txt:190 (find_package)


-- Configuring incomplete, errors occurred!
See also "C:/.conan/mocwshsk/1/CMakeFiles/CMakeOutput.log".
See also "C:/.conan/mocwshsk/1/CMakeFiles/CMakeError.log".
gRPC/1.8.3@inexorgame/stable:
gRPC/1.8.3@inexorgame/stable: ERROR: Package '4866a59ea942ec8def524b7f571b4b0f809525e9' build failed
gRPC/1.8.3@inexorgame/stable: WARN: Build folder C:/.conan\mocwshsk\1
ERROR: gRPC/1.8.3@inexorgame/stable: Error in build() method, line 81
        self.run('cmake {0}/{1} {2} {3}'.format(self.source_folder, self.folder, cmake.command_line, ' '.join(args)))

        ConanException: Error 1 while executing cmake C:/.conan\mocwshsk\1/grpc-1.8.3 -G "Visual Studio 14 2015 Win64" -DCONAN_LINK_RUNTIME="/MDd" -DCONAN_EXPORTED="1" -DCONAN_COMPILER="Visual Studio" -DCONAN_COMPILER_VERSION="14" -DBUILD_SHARED_LIBS="OFF" -DCMAKE_INSTALL_PREFIX="C:/.conan\4883jy36\1" -DCONAN_CXX_FLAGS="/MP4" -DCONAN_C_FLAGS="/MP4" -Wno-dev -DgRPC_INSTALL=ON -DgRPC_CARES_PROVIDER="module" -DCMAKE_INSTALL_PREFIX="C:/.conan\mocwshsk\1/install"

```

I added logs in `D:\Program Files\CMake\share\cmake-3.7\Modules\FindProtobuf.cmake`:
```
function(_protobuf_find_libraries name filename)
    ...
+  message("${name}_LIBRARY: " ${${name}_LIBRARY})
```
and got:
```
Protobuf_LIBRARY_RELEASE: Protobuf_LIBRARY_RELEASE-NOTFOUND
Protobuf_LIBRARY_DEBUG: Protobuf_LIBRARY_DEBUG-NOTFOUND
Protobuf_LIBRARY: Protobuf_LIBRARY-NOTFOUND
Protobuf_LIBRARIES: Protobuf_LIBRARY-NOTFOUND
Protobuf_LITE_LIBRARY_RELEASE: Protobuf_LITE_LIBRARY_RELEASE-NOTFOUND
Protobuf_LITE_LIBRARY_DEBUG: Protobuf_LITE_LIBRARY_DEBUG-NOTFOUND
Protobuf_LITE_LIBRARY: Protobuf_LITE_LIBRARY-NOTFOUND
Protobuf_LITE_LIBRARIES: Protobuf_LITE_LIBRARY-NOTFOUND
Protobuf_PROTOC_LIBRARY_RELEASE: Protobuf_PROTOC_LIBRARY_RELEASE-NOTFOUND
Protobuf_PROTOC_LIBRARY_DEBUG: Protobuf_PROTOC_LIBRARY_DEBUG-NOTFOUND
Protobuf_PROTOC_LIBRARY: Protobuf_PROTOC_LIBRARY-NOTFOUND
Protobuf_PROTOC_LIBRARIES: Protobuf_PROTOC_LIBRARY-NOTFOUND
```

`_protobuf_find_libraries(Protobuf protobuf)` can't find file `libprotobufd.lib`. 
If I rename 3 lib files to delete the 'd' suffix, the installation will succeed.

**Comment by Croydon** [comment]
It works on our CI fine on Linux with CMake 3.10: https://travis-ci.org/inexorgame/inexor-core/jobs/364482465#L2947

(Windows CI fails due to another non-related issue)

@jinq0123 Are you able to update CMake to a newer version? In the past there were several issues with the find package detections with diverse Conan packages.

**Comment by jinq0123** [comment]
CMake 3.11.0 succeeded.

**Comment by jinq0123** [comment]
Fixed in CMake 3.8.0:
```
SHA-1: d9190f3308fb67f01f061038a06006af73a77da1
* FindProtobuf: Search for debug library named with `d` suffix
```

**Comment by Croydon** [comment]
@jinq0123 Thanks for sharing.

We are only supporting CMake 3.9.2+ so this is fine from our perspective.

Feel free to open new issues if you run into further problems.
